main.py

```diff
@@ -427,6 +429,63 @@ def update():
     conn.close()
     return ''
 
+@app.route('/add_run3', methods=['POST'])
+def add_run3():
+    """
+    Endpoint to add a free text sample in run3 planning sheet
+    """
+    conn = sqlite3.connect('run3.db')
+    cursor = conn.cursor()
+    user_info = get_user_info(cursor)
+    if user_info['role'] == 'not a user':
+        logging.error('Could not find user %s, not doing anything', user_info)
+        return 'You are not a user of McM', 403
+
+    data = json.loads(request.data)
+
+    dataset_name = data['datasetname']
+    number_events = data['numberofevents']
+
+    pwg_list = []
+
+    sql_query = '''SELECT dataset
+                   FROM run3_samples
+                   WHERE  = ?'''
+
+    rows = cursor.execute(sql_query, [dataset_name])
+
+    #is the sample already in the list?
+    if rows is not None:
+        return ''
+
+    #input checks
+    if dataset_name is None or not number_events.replace(' ', '').isdigit():
+        return ''
+
+    #pwg checks: table is updated if there is at least 1 valid pwg
+    if 'pwginterested' in data:
+        for pwg_split in data['pwginterested'].replace(' ', '').split(','):
+            pwg = pwg_split.upper()
+            if pwg not in all_pwgs:
+                return 'Bad PWG %s' % (pwg), 400
+
+            else:
+                pwg_list.append(pwg)
+
+    if pwg_list is None:
+        #Nothing is added
+        return ''
+    else:
```

Comment:
You don't need an `else` after `return`.

```
if not pwg_list:
    # Nothing is added
    return 'No valid PWGs were found in request', 400

# This code is reached only if previous if is not entered
pwgs = ','.join(sorted(pwg_list))
...
```

On the other hand, I don't see how pwg_list can become None? Is this check needed?

Reply:
Agree!